### PR TITLE
Add reading from Kafka patches topic

### LINKED ISSUE #18: Implement Kafka subscription for patches
Subscribe to the topic where patches are being written to receive changes, and apply the changes to the HTML file.

This should probably just be a goroutine that starts whenever an instance of `ether` starts up.

### PR DESCRIPTION
**Pull request by igorveselinovic**
Resolves #18 

I realized that "eventually" this topic would actually contain all types of WebSocket messages in our protocol, because this topic is also being used to send messages between instances of `patches`, so the topic maybe shouldn't be `updates`, but `ws-messages` or something.

For the Kafka reading code, I kinda tried to emulate the HTTP handler pattern. So, if we needed `ether` to read from multiple topics, it could create multiple `kafkaReader` structs and call all their `Run()` methods with different `handler` functions. The main differences from the HTTP handlers is that we need multiple readers, but HTTP just needs one server, and the Kafka code runs the handlers sequentially (one-by-one), not concurrently.

The `CachedWriter` doesn't do any caching right now, but eventually it could store the file contents in its `files` map and only write back to the filesystem after a certain amount of time elapses. It just seemed like too big a design problem to solve in this PR (i.e., I'm lazy).

Also, I just copied over all the WebSocket protocol message structs from `patches` to save time (lazy, again).

### CODE REVIEW

File: kafka/message.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+// MessageType represents the possible Riht WebSocket protocol message types.
+type MessageType int
+
+// UpdateType represents the possible Update message subtypes.
+type UpdateType int
+
+const (
+	TypeInit      MessageType = 0
+	TypeUpdate    MessageType = 1
+	TypeAck       MessageType = 2
```
Comment (Thao-Tran) [inline kafka/message.go:18]: Add the new `Sync` message type.

File: filesystem/cachedWriter.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+				continue
+			}
+
+			newContent, okList := dmp.PatchApply(patches, string(content))
+			if !okList[0] {
+				log.Printf("Could not apply patch: %s", update.Patch)
+				continue
+			}
+
+			cw.directory.WriteFile(update.ConversationID, []byte(newContent))
```
Comment (Thao-Tran) [inline filesystem/cachedWriter.go]:
```suggestion
			err = cw.directory.WriteFile(update.ConversationID, []byte(newContent))
```
Right?

**Review by Thao-Tran — CHANGES_REQUESTED** [review]
Looks good. Just a couple small things.

File: filesystem/cachedWriter.go (hunk 2 above)
Comment (igorveselinovic) [reply filesystem/cachedWriter.go]: Right.

**Review by Thao-Tran — APPROVED** [review]

**Pull request merged**